docs/components/input-controller/_index.md

```diff
@@ -44,281 +44,475 @@ Once you've configured your input controller according to model type, you can wr
+    triggers := [1]input.EventType{input.ButtonPress}
 
-    // Get the controller from the robot.
-    myController, err := input.FromRobot(myRobotWithController, "my_controller")
+    // Get the controller's Controls.
+    controls, err := myController.Controls(context.Background(), nil)
 
-    // Log any errors that occur and exit if an error is found.
-    if err != nil {
-        logger.Fatalf("cannot get controller: %v", err)
+    // If the "ButtonStart" Control is found, register the function printStartTime to fire when "ButtonStart" has the event "ButtonPress" occur.
+    if slices.Contains(controls, Control.ButtonStart) {
+        err := controller.RegisterControlCallback(context.Background(), Control: input.ButtonStart, triggers, printStartTime, nil)
     }
+     else {
+        logger.Fatalf("Oops! Couldn't find the start button control! Is your controller connected?")
+    }
+}
+```
 
-    // Log an info message with the names of the different resources that are connected to your robot.
-    logger.Info("Resources:")
-    logger.Info(myRobotWithController.ResourceNames())
+{{% /tab %}}
+{{< /tabs >}}
 
-    // Get the controller's Controls.
-    controls, err := myController.Controls(context.Background(), nil)
+### Events
 
-    // Print out the controller's Controls.
-    logger.Info("Controls:")
-    logger.Info(resp)
+This method returns the current state of the controller as a map of [Event Objects](#event-object), representing the most recent event that has occured on each available [Control](#control-field).
 
-    // Log any errors that occur.
-    if err != nil {
-        logger.Fatal(err)
-    }
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+**Parameters:**
 
-    // Run the handleController function.
-    err := HandleController(myController)
+- `extra` [(Optional[Dict[str, Any]])](https://docs.python.org/library/typing.html#typing.Optional): Extra options to pass to the underlying RPC call.
+- `timeout` [(Optional[float])](https://docs.python.org/library/typing.html#typing.Optional): An option to set how long to wait (in seconds) before calling a time-out and closing the underlying RPC call.
 
-    // Delay closing your connection to your robot.
-    err = myRobotWithController.Start(ctx)
-    defer myRobotWithController.Close(ctx)
+**Returns:**
 
-    // Watch for errors.
-    if err != nil {
-        return err
-    }
+- `events` [(Dict[Control, Event])](https://docs.python.org/3/library/typing.html#typing.Dict): A dictionary mapping the most recent Event for each Control.
 
-    // Wait to exit mainWithArgs() until Context is Done.
-    <-ctx.Done()
+For more information, see the [Python SDK Docs](https://python.viam.dev/_modules/viam/components/input/input.html#Controller.get_events).
 
-    return nil
+```python {class="line-numbers linkable-line-numbers"}
+# Get the most recent Event for each Control.
+recent_events = await myController.get_events()
```

Comment:
Should this and other ones be
```suggestion
recent_events = await my_controller.get_events()
```
?
Or is it my_input_controller since sometimes it's that way below? Sorry if I'm confused

Reply:
Yeah, thanks for pointing this out -- I realized this one didn't have the `my_controller = Controller.from_robot(robot=robot, name="my_controller")` line etc so am fixing that~~ will switch to Python convention on this line too

Reply:
Fixed!